tools/mixxx-isolate.py

```diff
@@ -0,0 +1,496 @@
+#!/usr/bin/env python3
+import sys
+import os
+import platform
+import signal
+import re
+
+try:
+    # import cpuset
+    from cpuset.commands import set as ccset
+    from cpuset import cset
+    from cpuset.commands import proc
+    from cpuset.util import CpusetExists
+    from os import system
+
+except ImportError as e:
+    sys.stderr.write(e.msg)
+    sys.stderr.write("\nplease install cpuset\n")
+    sys.exit(1)
+
+
+import argparse
+import logging
+import subprocess
+from shutil import which
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("mixxx-isolate")
+
+
+def selectCpus(userCpu=-1):
+    """
+    Selects the most fitting cpu and it's ht companions.
+    Returns {"use": cpuid, "isolate": [cpuid, ...], "system": [cpuid, ...]}
+    """
+    cpuinfo = ""
+    with open("/proc/cpuinfo") as fp:
+        cpuinfo = fp.read(-1)
+
+    cpuToCore = {}
+    currentCpu = -1
+    lastCore = -1
+    for line in cpuinfo.splitlines():
+        if not line.strip():
+            continue
+        topic, data = line.split(":")
+        topic = topic.strip()
+        if topic == "processor":
+            currentCpu = int(data)
+        elif topic == "core id":
+            coreId = int(data)
+            cpuToCore[currentCpu] = coreId
+            lastCore = max(lastCore, coreId)
+
+    useCore = -1
+    if userCpu == -1:
+        useCore = lastCore
+    else:
+        useCore = cpuToCore.get(userCpu, -1)
+        if useCore == -1:
+            print("Selected CPU does not exist. Using autodetect.")
+            useCore = lastCore
+
+    useCpu = -1
+    isolate = []
+    system = []
+    for cpu, core in cpuToCore.items():
+        if core == useCore and useCpu == -1:
+            useCpu = cpu
+        elif core == useCore:
+            isolate.append(cpu)
+        else:
+            system.append(cpu)
+
+    return {"use": useCpu, "isolate": isolate, "system": system}
+
+
+def cpusetname(name):
+    if name and name[0] == "/":
+        return name
+    return "/%s" % name
+
+
+def deleteCpuset(name, debug):
+    try:
+        setInstance = cset.unique_set(name)
+    except cset.CpusetNotFound:
+        return True
+    # move processes until empty
+    processes = 1
+    while processes:
+        try:
+            log.debug(" Move processes from cpuset %s %s", setInstance, name)
+            proc.move(setInstance, "/", None, False)
+            processes = setInstance.tasks
+            log.debug(" Processes left: %s", processes)
+            break
+        except BrokenPipeError:
+            continue
```

Comment:
I feel like this is a significant risk to create an infinite loop. Shall we error out and break/abort here? Or maybe introduce a retry count limit